lib/src/main/java/edu/stanford/futuredata/macrobase/operator/Operator.java

```diff
@@ -1,5 +1,13 @@
 package edu.stanford.futuredata.macrobase.operator;
 
+/**
+ * Generic interface for operations. Should not modify input.
```

Comment:
Previously the semantics were to allow operators to destroy inputs; otherwise we have to copy everything on ever operation. What's the rationale for changing this?